daemon/execdriver/native/apparmor.go

```diff
@@ -60,12 +60,13 @@ profile {{.Name}} flags=(attach_disconnected,mediate_deleted) {
   deny /sys/firmware/efi/efivars/** rwklx,
   deny /sys/kernel/security/** rwklx,
 
+  # suppress ptrace denials when using 'docker ps' or using 'ps' inside a container
+  ptrace (trace,read) peer=docker-default,
```

Comment:
IIRC, older AppArmor versions can allow ptrace, but not granularly. That
is, you can allow ptrace, but cannot specify a peer or read/write/trace.

On Fri, Dec 4, 2015 at 12:00 PM, Jess Frazelle <notifications@github.com>
wrote:

> In daemon/execdriver/native/apparmor.go
> <https://github.com/docker/docker/pull/18393#discussion_r46704053>:
>
> > @@ -60,12 +60,13 @@ profile {{.Name}} flags=(attach_disconnected,mediate_deleted) {
> >    deny /sys/firmware/efi/efivars/** rwklx,
> >    deny /sys/kernel/security/** rwklx,
> >
> > +  # suppress ptrace denials when using 'docker ps' or using 'ps' inside a container
> > +  ptrace (trace,read) peer=docker-default,
>
> so the problem is this will still break distros like wheezy we need to
> change the version check to 2.8
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/18393/files#r46704053>.
>